service/worker/addsearchattributes/workflow.go

```diff
@@ -152,6 +155,20 @@ func (a *activities) AddESMappingFieldActivity(ctx context.Context, params Workf
 	return nil
 }
 
+func (a *activities) isRetryableError(err error) bool {
+	var httpStatusCode int
+	if err, isElasticErr := err.(*elastic.Error); isElasticErr {
+		httpStatusCode = err.Status
+	}
+
+	switch httpStatusCode {
+	case http.StatusBadRequest, http.StatusUnauthorized, http.StatusForbidden, http.StatusNotFound, http.StatusConflict:
+		return false
+	default:
+		return true
```

Comment:
This means we return true for all errors that aren't ES errors. Is that what we want?

Reply:
What do you mean "non-ES"? This error came from Elasticsearch and Elasticsearch uses http status codes to indicate error.

Reply:
I think I understand what you meant. Yes, non-ES errors are most likely some network failures and should be retryable. Also they might indicate some code bug (like bad formed url or missed required parameter). In this case it would be probably better not to retry but I don't know how to differentiate them. Generally, the idea is not to retry something that we know for sure is non-retryable and retry all the rest.